compile warnings for libsrc/posixio.c

```
posixio.c: In function ‘px_pgin’:
posixio.c:343:11: warning: unused variable ‘p’ [-Wunused-variable]
     void *p = vp;
           ^
posixio.c:342:13: warning: unused variable ‘bytes_xfered’ [-Wunused-variable]
     ssize_t bytes_xfered = 0;
             ^~~~~~~~~~~~
posixio.c:341:12: warning: unused variable ‘read_count’ [-Wunused-variable]
     size_t read_count = 0;
            ^~~~~~~~~~

```

--- a/libsrc/posixio.c
+++ b/libsrc/posixio.c
@@ -338,9 +338,6 @@ px_pgin(ncio *const nciop,
 {
 	int status;
 	ssize_t nread;
-    size_t read_count = 0;
-    ssize_t bytes_xfered = 0;
-    void *p = vp;
 #ifdef X_ALIGN
 	assert(offset % X_ALIGN == 0);
 	assert(extent % X_ALIGN == 0);

